### DISCUSSION TITLE
Abstimmanzeige bzw. Feldstärkenanzeige möglich ?

### DISCUSSION THREAD

**Discussion by Zurrmaxe** [hoylabs/OpenDTU-OnBattery#823] [Ideas]
Hallo zusammen
Da bei uns auf dem Lande alles etwas weiter "auseinander" liegt, habe ich manchmal Probleme, den richtigen "Standort" von OpenDTU , dem Wechselrichter und dem WLAN-Netz zu finden. 
Hilfreich wäre eine Abstimmanzeige bzw. Feldstärkenanzeigen (z.B. Balken links und Rechts im Display /Wechselrichter/CMT2300A/WLAN) wünschenswert, damit die externen Antennen und die openDTU optimal positioniert werden kann.  Gleichzeitig könnte man auch erkennen, ob man überhaupt etwas empfängt oder etwas defekt ist.
lg Alfred

**Comment by spcqike** [comment]
Ob es dafür eine dauerhafte Anzeige benötigt? Die Installation wird doch in der Regel nur einmal gemacht. 

Soweit ich weiß sieht man in der Konsole einen dB wert für die Kommunikation. Aber ob dieser reel ist und tatsächlich Rückschlüsse auf die Empfangsqualität zulässt? Kann ich nicht abschätzen. 

Zusätzlich wäre es ein übergeordnetes openDTU Thema und kein onBattery spezifisches. Daher gehört es eher in den upstream.

**Comment by schlimmchen** [comment]
Alles richtig, was spcqike schreibt. Der Aufwand zur Implementierung übersteigt meiner Meinung nach den (einmaligen) Nutzen.

Ich möchte noch ergänzen, dass du in er Web App unter Info -> Netzwerk sowohl einen Qualitätsindikator als auch den RSSI des WLANs ablesen kannst.

Die Position relativ zu den Hoymiles ist meines Erachtens nach in der Praxis kaum so schlecht wählbar, dass die Kommunikation zu schlecht würde. Die Qualität der Kommunikation mit den Hoymiles kannst du abschätzen indem du den Live View beobachtest: Kommen weiterhin sehr regelmäßig neue Daten vom Inverter, dann ist alles dufte. Stell das Polling-Intervall sehr klein ein für diese Beobachtung, dann fallen Ausfälle schnelle auf.

**Comment by Manos1966** [comment]
Ich kenne das Problem...

Ist es ein WiFi-Problem, oder ein NRF24-Problem? Ich tippe auf WiFi...
Wie spcqike & schlimmchen gesagt haben, dies ist eher ein openDTU Thema.

Aber, es gibt ein paar Loesungen die vielleicht helfen koennten (ohne Programmier-Arbeit):

1 Wechselrichter-Kommunikation: NRF24 (oder CMT2300A)
Ich gehe davon aus, du hast eine nRF24L01+ (oder CMT2300A) Platine mit SMA Antenna = **externe Antenne.**
falls nicht, unbedingt Platine wechseln!
https://www.opendtu.solar/hardware/nrf24l01plus/#pictures
...und den Kapazitator in der Nähe der Platine nicht vergessen.

2 WiFi Thema (ESP32)
Die meisten ESP32 haben eine eingebaute WiFi Antenne von fragwürdiger Qualität.
Es gibt aber ESP32 Platinen die eine externe WiFi Antenne aufnehmen koennen:
![Capture](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/1d55e332-5fd0-4998-bb9d-3be3364f5393)

und die externe Antenne kannst du platzieren wo es am besten passt.

**Comment by Zurrmaxe** [comment]
Hallo zusammen
Leider hat man nicht immer das Glück, dass man die Solaranlage auf seinem Hausdach hat.
Bei mir und bei anderen hier in der Gegend werden die Solarplatten oftmals auf die reichlich vorhandenen Stallungen und Schuppen, fernab von erreichbarem WLAN, angebracht. 
Bei mir musste ich deshalb die DTU auf halben Wege zwischen der Solaranlage und WLAN -Netz anbringen.
Da wäre es schön, wenn man direkt in dem Display die Signalstärke hätte. 
Diese bräuchte auch nur einmalig beim Ausrichten zuschaltbar zu sein. 
Nachdem es nicht geklappte hatte, habe ich es mit externen Antennen versucht. 
Davon habe ich hier als Amateurfunker genug herumliegen.

Letzlich hatte sich herausgestellt, dass die bei den Bausätzen mitgelieferten Antennen  billiger China-Schrott sind und garnicht für die DTU (863MHz) geeignet waren.  Hierzu habe ich dann auch etwas geschrieben. 

Vermutlich ist dieser Antennen-Schrott auch für viele aufgetretenen "Störungen/Problemen" bei anderen Usern verantwortlich.  

Nachdem ich die Antennen ensprechend verändert und abgeglichen hatte, funktionierte auch alles Bestens.

mfg. Alfred

![Antennen](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/164932724/7b1d2872-1e99-4e4e-bb2d-0e39ab8ecbc1)
irte auch alles.

**Reply by Manos1966** [reply]
Danke @Zurrmaxe 

aus Neugier, welche Länge braucht man für WiFi 2,4GHz (und 5GHz)?
Weil ich befürchte, es existiert auch eine WiFi-Herausforderung

![Untitled](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/14f2025d-f42f-408a-81f6-da4b0142aa1c)

**Comment by Manos1966** [comment]
Der Vollständigkeit halber, muss ich noch etwas addieren, für alle die eine OpenDTUonBattery mit Batterie (und dadurch VICTRON MPPT oder Huwaei Netzteil) nutzen:
Die OpenDTUonBattery befindet sich sehr nahe an die Batterie und die Hoymiles.
In einen solchen Fall ist eine gute Antenne von *NACHTEIL* und ein "nRF24L01+ Module with PCB antenna" ist besser 
![20230828_011807 - Copy](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/6ba5e885-46e9-49f4-aa2f-198fbbd035ee)
(der  CMT2300A kommt leider ausschließlich mit externe Antenne)

Ansonsten existiert das Risiko 
#392 Request retransmit: 4 / Nothing received, resend whole request

**Comment by Zurrmaxe** [comment]
Grundsätzlich verseuchen alle Schaltnetzteile und dazu gehören im weitesten Sinne auch MPPT-Schaltunge, die direkte Umgebung mit einem Störnebel.  Irgendwelkche Empfänger sollte man daher nicht in unmittelbarer Nähe zu solchen Schalteungen oder Geräte aufstellen, weil der Störnebel das Nutrzsignal überlagern kann.
Wie die Antennen aus China 2,4 und 5 GHz machen wollen, kann ich mir technisch wohl vorstellen, dürfte aber für den Preis nicht gehen
Dazu müssten man mehrere kleine "Antennenstücke" mit Spulen und ggf. Kapazitäten verbinden um auf eine gestockte 3db antenne zu kommen.
Wie hat man die 3dB ( keine 3dBi !!) gemessen ? Gegen Grund ? Isotopenstrahler ? Keine Ahnung.  
Ich vermute darin auch nur einen mehr oder weniger langen Draht.
Eine Lambda/4 für 2.4GHz ist fast auch als Lambda/2 für 5GHz nutzbar.  Um das genau festzustellen, müsste man die Teile ausmessen.
Alfred

**Reply by schlimmchen** [reply]
> Eine Lambda/4 für 2.4GHz ist fast auch als Lambda/2 für 5GHz nutzbar.

Das ist nach allem, was ich weiß (leider keinen Plan von Funk), die Idee.

> Grundsätzlich verseuchen alle Schaltnetzteile und dazu gehören im weitesten Sinne auch MPPT-Schaltunge

Das ist nachvollziehbar, aber erklärt mir nicht, warum manche (mich eingeschlossen) mit Antennendämpfungsgliedern eine "bessere" (stabilere) Kommunikation zwischen Homyiles WR und der OpenDTU-OnBattery hinbekommen. Passt das trotzdem zusammen? Ich find den Gedanken interessant, weil ich mir noch immer nicht erklären kann, warum die (sehr kurze) Funktstrecke tagesformabhängig zu sein scheint. Vielleicht weil der MPPT abhängig von der Leistung mehr oder weniger stört?

**Reply by Zurrmaxe** [reply]
Die verwendeten Funkempfänger haben keine Selektion oder abstimmbare Vorkreise wie ein Funkgerät. Sie sind sehr breitbandig gehalten, damit sie für viele Anwendungen benutzt werden können. Wenn man sie schmalbandig macht, müsste der Hersteller für fast jedem 1 Mhz  -Bereich einen speziellen Chip bauen. Viel zu teuer...
Durch das Vorschalten von Dämpfungsgliedern wird naturgemäß alles reduziert. Nutz-und Störsignal. 
10 db z.B ist ein Zehntel des ursprunglichen Signales. Anscheinend reicht das für einen selektiven Empfang aus.
Würde man den Empfänger, statt neben dem "Störer", 10 Meter entfernt aufstellen, hätte man vermutlich den gleichen Effekt.
Falls man Zugriff auf ein entsprechendes Messgerät (Spectrumanalyser) hat, kann man sich den HF-Müll genauer ansehen und die Ursache bestiummt finden. Ansonsten bleibt nur die übliche Methode : Try and Error

**Reply by Zurrmaxe** [reply]
![10db Kopie](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/164932724/d8994e3d-77af-4d1b-b826-6ed42e5f981f)

**Comment by Manos1966** [comment]
> Eine Lambda/4 für 2.4GHz ist fast auch als Lambda/2 für 5GHz nutzbar

OK also 30mm fuer WiFi (2,4GHz & 5GHz) & NRF24 und 86mm (corrected) for CMT2300 
klingt einfach 👍

**Reply by spcqike** [reply]
Sind es nicht 86mm? Die Antenne ist doch keine 86cm aka fast 1m lang (:

**Comment by Zurrmaxe** [comment]
Stimmt, ca. 86 mm
